geps/gep-2162.md

```diff
@@ -0,0 +1,114 @@
+# GEP-2162: Supported features in GatewayClass Status
+
+* Issue: [#2162](https://github.com/kubernetes-sigs/gateway-api/issues/2162)
+* Status: Provisional
+
+## TLDR
+
+This GEP proposes to enhance the [GatewayClassStatus](https://github.com/kubernetes-sigs/gateway-api/blob/f2cd9bb92b4ff392416c40d6148ff7f76b30e649/apis/v1beta1/gatewayclass_types.go#L185) to include a list of gw-api features supported by the installed Gateway Class. 
+
+## Goals
+
+* Enable users easily see what features the implementation(gateway class) support
+* Improving UX and potentially provide clear errors when a user is trying to use a feature the installed GWC doesn't support
+
+## Non-Goals
+
+* Validate correctness of supported features published by the implementation.
+
+## Introduction
+
+The current [GatewayClassStatus](https://github.com/kubernetes-sigs/gateway-api/blob/f2cd9bb92b4ff392416c40d6148ff7f76b30e649/apis/v1beta1/gatewayclass_types.go#L185 is only used to store conditions the controller publishes.
+
+Partnered with the [Conformance Profiles](https://github.com/kubernetes-sigs/gateway-api/blob/main/geps/gep-1709.md) work, we want to:
+
+1. Enable users to easily see what features an implementation supports
+1. Improve UX and potentially provide clear errors when a user is trying to use an unsupported feature
```

Comment:
1. Automatically run conformance tests based on the supported features populated in GatewayClass status

Reply:
I am not sure there is consensus around that yet. I think it needs more discussion. 

@shaneutt IIRC you had different ideas but do let me know if I am misremember and your are good to go with it.